hack/verify-go-clean.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+#####
+#  Used to ensure that running `go mod tidy` results in a non-dirty repository.
+#  Used to verify that dependency changes have been cleaned
+#####
+
+set -o nounset
+set -o pipefail
+# intentionally not setting 'set -o errexit' because we want to print custom error messages
+
+# versions of tools
+echo "go $(go version)"
```

Comment:
Nitpick: `go version` returns strings like "go version go1.15.2 darwin/amd64" so the additional "go" here is redundant.
```suggestion
echo "$(go version)"
```